ui/src/components/AppDetailedView.js

```diff
@@ -29,18 +30,37 @@ class AppDetailedView extends React.Component {
     this.unblockHistory();
   }
 
+  onSaveApp () {
```

Comment:
```suggestion
  onSaveApp = () => {
```
this will bind the `this` to this component when called, otherwise you will have to bind it in a constructor here, ie
```
this.onSaveApp = this.onSaveApp.bind(this);
```

Reply:
It doesn't seem that I need to that binding in the constructor - what is there currently works.

Reply:
Yes, in this case it seems the context is handled correctly as it's handled by passing the entire handler as a prop to `AppDetailedToolbar`. I would still suggest explicitly binding `this` though to avoid potential future bugs if `onSaveApp` is called from a different context than `AppDetailedView`

Reply:
Thanks for the explanation :+1:

Reply:
Updated with your suggested change.